### PR TITLE
Linear Gaussain CPD

### PR DESCRIPTION
**Pull request by yashu-seth**

### CODE REVIEW

**Comment by ankurankan** [comment]
@yashu-seth There are merge conflicts in this. And please let me know when this is ready for review.

File: pgmpy/factors/continuous/LinearGaussianCPD.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+from scipy.stats import multivariate_normal
+
+from pgmpy.extern import six
+from pgmpy.factors import ContinuousFactor
+
+
+class LinearGaussianCPD(ContinuousFactor):
+    u"""
+    For, X -> Y the Linear Gaussian model assumes that the mean
+    of Y is a liner function of mean of X and the variance of Y does
```
Comment (ankurankan) [inline pgmpy/factors/continuous/LinearGaussianCPD.py]: @yashu-seth typo: liner -> linear

File: pgmpy/factors/continuous/LinearGaussianCPD.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    not depend on X.
+
+    For example,
+    p(Y|X) = N(-2x + 0.9 ; 1)
+
+    Let Y be a continuous variable with continuous parents
+    X1 ............ Xk . We say that Y has a linear Gaussian CPD
+    if there are parameters β0,.........βk and σ2 such that,
+
+    p(Y |x1.......xk) = N(β0 + β1 + ......... + βk ; σ2)
```
Comment (ankurankan) [inline pgmpy/factors/continuous/LinearGaussianCPD.py]: Shouldn't this be: `p(Y | x1, x2, ...., xk) = N(\beta_0 + x1 \beta_1 + ... + xk \beta_k ; \sigma_2)` ?

File: pgmpy/factors/continuous/LinearGaussianCPD.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    In vector notation,
+
+    p(Y |x) = N(β0 + β.T * x ; σ2)
+
+    """
+    def __init__(self, variable, beta_not, variance, evidence=None, beta_vector=None):
```
Comment (ankurankan) [inline pgmpy/factors/continuous/LinearGaussianCPD.py]: Change `beta_not` to `beta_0`

File: pgmpy/factors/continuous/LinearGaussianCPD.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        --------
+        >>> import numpy as np
+        >>> from pgmpy.factors import LinearGaussianCPD
+        >>> cpd = LinearGaussianCPD('Y', 0.2, 9.6, ['X1', 'X2', 'X3'], [-2, 3, 7])
+        >>> copy_cpd = cpd.copy()
+        >>> copy_cpd.variable
+        'Y'
+        >>> copy_cpd.evidence
+        ['X1', 'X2', 'X3']
+        >>>
```
Comment (ankurankan) [inline pgmpy/factors/continuous/LinearGaussianCPD.py]: Delete this line.

**Comment by ankurankan** [comment]
@yashu-seth Please add tests for this.

**Comment by yashu-seth** [comment]
@ankurankan I have added `LinearGaussainBayesianNetwork`. Please have a look. Meanwhile I will add tests for `LinearGaussianCPD`.

**Pull request merged**